web_src/js/utils.js

```diff
@@ -97,3 +97,8 @@ export function prettyNumber(num, locale = 'en-US') {
   const {format} = new Intl.NumberFormat(locale);
   return format(num);
 }
+
+// parse a URL, either relative '/path' or absolute 'https://localhost/path'
+export function parseUrl(str) {
+  return new URL(`${str.startsWith('http') ? '' : window.location.origin}${str}`);
```

Comment:
FYI, this should ideally use the `base` option of `URL()` instead of concatenating, but I [hit a bug in happy-dom](https://github.com/capricorn86/happy-dom/issues/569) which prevents it.

Reply:
I've done the switch to `jsdom`, no workarounds necessary with it anymore as it sets `window.location` to `http://localhost`.